include/boost/math/tools/roots.hpp

```diff
@@ -354,7 +354,12 @@ namespace detail {
          BOOST_MATH_INSTRUMENT_VARIABLE(denom);
          BOOST_MATH_INSTRUMENT_VARIABLE(num);
 
-         if ((fabs(num) < 1) && (fabs(denom) >= fabs(num) * tools::max_value<T>()))
+         // denom/num overflows if:
+         //     |denom| >= |num| * max_value
+         // RHS may overflow on Apple M1, so rearrange:
+         //     |denom| * 1/max_value >= |num|
+         constexpr T inv_max_value = 1.0 / tools::max_value<T>();
```

Comment:
```suggestion
         const T inv_max_value = 1.0 / tools::max_value<T>();
```

Reply:
Even better make that `static const` and then it's evaluated just the once.

Reply:
I wanted `static constexpr`, but wasn't sure that is supported everywhere we want, and indeed, it appears even `constexpr` is problematic.  Will implement this right away